rir/src/compiler/opt/constantfold.cpp

```diff
@@ -131,11 +131,21 @@ void Constantfold::apply(RirCompiler& cmp, ClosureVersion* function,
                 next = bb->remove(ip);
             });
 
-            if (auto isObj = IsObject::Cast(i)) {
-                if (!isObj->arg<0>().val()->type.maybeObj()) {
+            if (auto isTest = TypeTest::Cast(i)) {
+                if (isTest->testFor == TypeTest::Object &&
+                    !isTest->arg<0>().val()->type.maybeObj()) {
                     i->replaceUsesWith(False::instance());
                     next = bb->remove(ip);
                 }
+
+                if (isTest->testFor == TypeTest::EnvironmentStub) {
+                    auto environment = MkEnv::Cast(isTest->arg<0>().val());
```

Comment:
I don't see any invariant in the isTest instruction, that guarantees that this argument is a `MkEnv`.

Either add invariants to the stubtest, or test here, ie. `if (auto environment = MkEnv::Cast(...)) {`